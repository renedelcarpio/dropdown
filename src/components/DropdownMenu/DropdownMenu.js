import './DropdownMenu.scss';

import DropdownItem from '../DropdownItem/DropdownItem';
import React from 'react';
import { useState } from 'react/cjs/react.development';

const DropdownMenu = ({ setSelectedOption }) => {
	const options = ['id', 'reference_name', 'nit', 'name'];

	const [isOpen, setIsOpen] = useState(false);

	/**
	 * This function change the state true to show the list of DropdownItems or false to hide this
	 */
	const handleToggle = () => {
		setIsOpen(!isOpen);
	};

	/**
	 * This function set the innerHtml of the target to the selectedOption and change the isOpen boolean to false
	 * @param {target} event Make reference to the event
	 */
	const onOptionSelected = ({ target }) => {
		setSelectedOption(target.innerHTML);
		setIsOpen(false);
	};

	return (
		<>
			<div
				className={isOpen ? 'dropdown__menu--open' : 'dropdown__menu'}
				onClick={handleToggle}
			>
				Dropdown Menu
			</div>
			{isOpen && (
				<div className='dropdown__items'>
					{options.map((option, index) => (
						<DropdownItem onClick={onOptionSelected} key={`${option}${index}`}>
							{option}
						</DropdownItem>
					))}
				</div>
			)}
		</>
	);
};

export default DropdownMenu;
