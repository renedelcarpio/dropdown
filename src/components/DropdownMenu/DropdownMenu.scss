@import '../../index.scss';

.dropdown__menu {
  border: 0.2rem solid $color-primary;
  border-radius: 1rem;
  color: $color-secondary;
  font-size: 1.5rem;
  font-weight: bold;
  padding: 1.2rem 2.5rem;
  transition: all 0.3s;

  &:hover {
    background-color: $color-primary;
    color: $color-light;
  }
}

.dropdown__menu--open {
  background-color: $color-primary;
  border: 0.2rem solid $color-primary;
  border-radius: 1rem;
  color: $color-light;
  font-size: 1.5rem;
  font-weight: bold;
  padding: 1.2rem 2.5rem;
}

.dropdown__items {
  position: absolute;
  width: 25rem;
}