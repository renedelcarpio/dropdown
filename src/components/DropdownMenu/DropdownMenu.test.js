import DropdownMenu from './DropdownMenu';
import { mount } from 'enzyme';

describe('Tests in DropdownMenu component', () => {
	const wrapper = mount(<DropdownMenu />);

	test('should match with snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should not exist before click', () => {
		expect(wrapper.find('.dropdown__items').exists()).toBe(false);
	});
});
