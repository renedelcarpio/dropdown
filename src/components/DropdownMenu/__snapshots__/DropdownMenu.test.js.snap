// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in DropdownMenu component should match with snapshot 1`] = `
<div
  className="dropdown__menu"
  onClick={[Function]}
>
  Dropdown Menu
</div>
`;
